release-notes/Cabal-3.14.0.0.md

```diff
@@ -0,0 +1,152 @@
+### Significant changes
+
+- Neutral field to add files to sdist [#8817](https://github.com/haskell/cabal/issues/8817) [#10107](https://github.com/haskell/cabal/pull/10107)
+
+  Adds the `extra-files` field to the cabal file specification. This is like
+  the other `extra-*` fields in that it is copied with the `sdist` command,
+  except there are no other semantics. Compare to:
+
+  * `extra-source-files`: Tracked by `cabal build`.
+
+  * `extra-doc-files`: Copied by Haddock to the html directory.
+
+### Other changes
+
+- Include package version when passing `--promised-dependency` flag [#10166](https://github.com/haskell/cabal/issues/10166) [#10248](https://github.com/haskell/cabal/pull/10248)
+
+  The `--promised-dependency` flag now expects an argument in the format
+
+  ```
+  NAME-VER[:COMPONENT_NAME]=CID
+  ```
+
+  rather than
+
+  ```
+  NAME[:COMPONENT_NAME]=CID
+  ```
+
+- Add support for building profiled dynamic way [#4816](https://github.com/haskell/cabal/issues/4816) [#9900](https://github.com/haskell/cabal/pull/9900)
+
+  Add support for profiled dynamic way
+
+  New options for `cabal.project` and `./Setup` interface:
+
+  * `profiling-shared`: Enable building profiling dynamic way
+  * Passing `--enable-profiling` and `--enable-executable-dynamic` builds
+    profiled dynamic executables.
+
+  Support for using `profiling-shared` is guarded behind a constraint
+  which ensures you are using `Cabal >= 3.13`.
+
+  In the cabal file:
+
+  * `ghc-prof-shared-options`, for passing options when building in
+    profiling dynamic way
+
+- Working directory support for `Cabal` [#9702](https://github.com/haskell/cabal/issues/9702) [#9718](https://github.com/haskell/cabal/pull/9718)
+
+  The `Cabal` library is now able to handle a passed-in working directory, instead
+  of always relying on the current working directory of the parent process.
+
+  In order to achieve this, the `SymbolicPath` abstraction was fleshed out, and
+  all fields of `PackageDescription` that, if relative, should be interpreted
+  with respect to e.g. the package root, use `SymbolicPath` instead of `FilePath`.
+
+  This means that many library functions in `Cabal` take an extra argument of type
+  `Maybe (SymbolicPath CWD (Dir "Package"))`, which is an optional (relative or
+  absolute) path to the package root (if relative, relative to the current working
+  directory). In addition, many functions that used to manipulate `FilePath`s now
+  manipulate `SymbolicPath`s, require explicit conversion using e.g. `getSymbolicPath`.
+
+  To illustrate with file searching, the `Cabal` library defines:
+
+  ```haskell
+  findFileCwd
+    :: forall dir1 dir2 file
+     . Verbosity
+    -> Maybe (SymbolicPath CWD (Dir dir1))
+
+    -> [SymbolicPath dir1 (Dir dir2)]
+
+    -> RelativePath dir2 File
+
+    -> IO (SymbolicPath dir1 File)
+  ```
+
+  See Note [Symbolic paths] in `Distribution.Utils.Path` for further information
```

Comment:
```suggestion
  See Note "Symbolic paths" in `Distribution.Utils.Path` for further information
```

(or `Symbolic paths`)

Reply:
Not sure that works: it's using the GHC embedded notes convention, as such the note itself includes the brackets.

Reply:
Maybe ` [Symbolic paths]` then?

As it is now, it looks like we forgot the URL part in a Markdown link.

Reply:
I think if you dislike the GHC Note convention, the right thing to do is to get the author to make a new PR replacing it with something acceptable.

Reply:
Sorry I meant `Note [Equality-constrained types]`

(i.e. the very same thing that there is now, just sandwiched in ``)